pandas/core/indexes/datetimelike.py

```diff
@@ -222,6 +221,18 @@ class DatetimeIndexOpsMixin(DatetimeLikeArrayMixin):
     _resolution = cache_readonly(DatetimeLikeArrayMixin._resolution.fget)
     resolution = cache_readonly(DatetimeLikeArrayMixin.resolution.fget)
 
+    def _shallow_copy(self, values=None, **kwargs):
+        if isinstance(values, list):
+            # reached via Index.insert
+            assert len(values) == 0
+            values = np.array([], dtype='i8')
+
+        # unwrap for case where e.g. _get_unique_index passes an instance
+        #  of own class instead of ndarray
+        values = getattr(values, '_data', values)
```

Comment:
Can you do this with an actual check that it is an index?

Reply:
Sure.  (Ideally I'd like to make _get_unique_index pass the "correct" thing, but that can wait for another day)